dhall/src/Dhall/Import.hs

```diff
@@ -337,7 +337,7 @@ instance Show CannotImportHTTPURL where
 {-|
 > canonicalize . canonicalize = canonicalize
 
-> canonicalize (a <> b) = canonicalize a <> canonicalize b
+> canonicalize (a <> b) = canonicalize (canonicalize a <> canonicalize b)
```

Comment:
Can you turn these laws into property tests similar to what we have in [`Dhall.Map`](https://github.com/dhall-lang/dhall-haskell/blob/master/dhall/src/Dhall/Map.hs#L122)?

Reply:
Hmm, could you point me to where `prop> ...` annotations are defined? Though I don't expect it to be possible to give property tests for typeclass laws, since they would have to be tested on each instance.

Reply:
> Hmm, could you point me to where `prop> ...` annotations are defined?

https://github.com/dhall-lang/dhall-haskell/blob/e044b4ab6873cf95ae0ca18461960998d64f0f65/dhall/src/Dhall/Map.hs#L122

https://github.com/dhall-lang/dhall-haskell/blob/e044b4ab6873cf95ae0ca18461960998d64f0f65/dhall/src/Dhall/Map.hs#L129-L130

The Arbitrary instance is defined at the end of the file:

https://github.com/dhall-lang/dhall-haskell/blob/e044b4ab6873cf95ae0ca18461960998d64f0f65/dhall/src/Dhall/Map.hs#L667-L673

> Though I don't expect it to be possible to give property tests for typeclass laws, since they would have to be tested on each instance.

Oops, my mistake. Yeah, we should test these instances – there are only 5 in total. Could be done in the testsuite though.

Reply:
I think I'll leave that as "future work" xD

Thanks for the pointers though, I didn't know you could specify quickcheck properties in haddock comments!